addons/website_sale_mass_mailing/controllers/main.py

```diff
@@ -0,0 +1,17 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.http import request, route
+from odoo.addons.website_mass_mailing.controllers import main as website_mass_mailing
+from odoo.addons.website_sale.controllers import main
+
+
+class WebsiteSale(main.WebsiteSale, website_mass_mailing.MassMailController):
+
+    @route()
+    def address(self, **kw):
+
+        if 'submitted' in kw and request.httprequest.method == "POST":
+            if kw.get('newsletter'):
+                self.subscribe(request.website.newsletter_id, kw['email'], 'email')
+
+        return super().address(**kw)
```

Comment:
That should do the trick :) See https://stackoverflow.com/questions/14743787/python-superclass-self-method-vs-superparent-self-method
```suggestion
        return super(WebsiteSale, self).address(**kw)
```